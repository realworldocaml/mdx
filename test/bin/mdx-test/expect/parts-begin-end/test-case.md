Mdx can also understand ocaml code blocks:


```ocaml file=parts-begin-end.ml,part=toto
# let x = 3;;
val x : int = 3
# let y = 4;;
val y : int = 4
# print_int x;;
3
- : unit = ()
# print_int y;;
4
- : unit = ()
```

```ocaml file=parts-begin-end.ml,part=z_zz
```

```ocaml file=parts-begin-end.ml,part=4-2
```

```ocaml file=parts-begin-end.ml
```

```ocaml
# let x = 2;;
val x : int = 2
# print_int x;;
3
- : unit = ()
```

```ocaml file=parts-begin-end.ml,part=indented
```
