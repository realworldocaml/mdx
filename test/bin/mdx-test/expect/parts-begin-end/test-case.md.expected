Mdx can also understand ocaml code blocks:


```ocaml file=parts-begin-end.ml,part=toto
let x = 34
let f = 42.3
let s = "toto"
let f x u = u x

let () =
  print_int x;
  print_float f
;;
```

```ocaml file=parts-begin-end.ml,part=z_zz
let () =
  print_string s
;;
```

```ocaml file=parts-begin-end.ml,part=4-2
let () =
  f x print_int;
```

```ocaml file=parts-begin-end.ml
let () =
  ();
  ()
;;

let x = 34
let f = 42.3
let s = "toto"
let f x u = u x

let () =
  print_int x;
  print_float f
;;

p
let () =
  print_string s
;;


let () =
  f x print_int;
  ()

let () =
  let () = fooooooooooooooooooooooooooooooooooooooooooo in
  if not fooooooooo then foooooooooooo
```

```ocaml
# let x = 2;;
val x : int = 2
# print_int x;;
2
- : unit = ()
```

```ocaml file=parts-begin-end.ml,part=indented
  let () = fooooooooooooooooooooooooooooooooooooooooooo in
  if not fooooooooo then foooooooooooo
```
